src/mo_frontend/typing.ml

```diff
@@ -1150,7 +1150,18 @@ and infer_exp'' env exp : T.typ =
         { env with async = C.NullCap; in_actor = true }
       else env
     in
-    infer_obj env' obj_sort.it dec_fields exp.at
+    let t = infer_obj env' obj_sort.it dec_fields exp.at in
+    begin match typ_opt with
+      | Some typ ->
+        let t' = check_typ env' typ in
+        if not (T.sub t t') then
+          local_error env exp.at "M0192"
+            "body of type%a\ndoes not match expected type%a"
```

Comment:
```suggestion
            "body of type%a\ndoes not match expected type%a"
```
Should we call this the `annotated` (not expected) type? Or maybe the `expected interface` type? Then again, I don't we thing we are that informative for the class type annotation either.

Reply:
Right. Either this way, or we change both.